DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -156,23 +158,33 @@ internal class MarshallingGenerators
                 case { ManagedType: { SpecialType: SpecialType.System_String } }:
                     return CreateStringMarshaller(info, context);
 
-                case { ManagedType: { TypeKind: TypeKind.Delegate }, MarshallingAttributeInfo: NoMarshallingInfo or MarshalAsInfo { UnmanagedType: UnmanagedType.FunctionPtr } }:
+                case { ManagedType: { TypeKind: TypeKind.Delegate }, MarshallingAttributeInfo: NoMarshallingInfo or MarshalAsInfo(UnmanagedType.FunctionPtr) }:
                     return Delegate;
 
-                case { MarshallingAttributeInfo: BlittableTypeAttributeInfo _ }:
+                case { MarshallingAttributeInfo: BlittableTypeAttributeInfo }:
                     return Blittable;
 
                 // Marshalling in new model
                 case { MarshallingAttributeInfo: NativeMarshallingAttributeInfo marshalInfo }:
                     return Forwarder;
 
                 // Simple marshalling with new attribute model, only have type name.
-                case { MarshallingAttributeInfo: GeneratedNativeMarshallingAttributeInfo { NativeMarshallingFullyQualifiedTypeName: string name } }:
+                case { MarshallingAttributeInfo: GeneratedNativeMarshallingAttributeInfo(string nativeTypeName) }:
```

Comment:
I don't understand the `string nativeTypeName` here. Why is that there?

Reply:
This statement has the same meaning as the previous version, but instead of using the property deconstruction syntax, it uses the positional syntax that matches the record's constructor (well, technically the auto-generated `Deconstruct` method that matches the primary constructor). So this case matches a `GeneratedNativeMarshallingAttributeInfo` attribute that was created with any `string`. That `string` is then stored in the `nativeTypeName` variable.

Reply:
I think I understood that actually - yay for being in on the new C# stuff. What I don't see is where `nativeTypeName` is being used, that is my main confusion.

Reply:
It's not being used today. I originally added `nativeTypeName` when I was starting work on the non-blittable struct marshaller. It will be used when we support generating code for non-blittable structs to enable us to have both source generators working without a dependency between the two (which isn't currently allowed). `nativeTypeName` will be a pre-calculated name for what the struct marshalling source generator will name the native type.

Reply:
I can remove it and just match on `GeneratedNativeMarshallingAttribute` until we implement support if you want.

Reply:
Not necessary. Wanted to make sure we knew why.